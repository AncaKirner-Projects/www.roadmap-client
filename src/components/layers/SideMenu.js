import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import IconButton from '@material-ui/core/IconButton';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import styled from 'styled-components';

import Categories from '../main/Categories';

const TypoStlyled = styled(Typography)`
  padding: 10px 0px 0px 10px;
  text-decoration: none;
`;

class SideMenu extends Component {
  state = {
    open: true
  };

  render() {
    const { classes, theme } = this.props;
    const { open } = this.state;
    return (
      <Drawer
        variant="permanent"
        className={classNames(classes.drawer, {
          [classes.drawerOpen]: open,
          [classes.drawerClose]: !open
        })}
        classes={{
          paper: classNames({
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open
          })
        }}
        open={open}
      >
        <div className={classes.toolbar}>
          <IconButton>
            {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
          </IconButton>
        </div>
        <Divider />
        <TypoStlyled variant="h6" to="/" component={Link}>Categories</TypoStlyled>
        <List>
          <Categories />
        </List>
      </Drawer>
    );
  }
}

SideMenu.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    categories: state.categories
  };
}

export default connect(mapStateToProps)(SideMenu);
