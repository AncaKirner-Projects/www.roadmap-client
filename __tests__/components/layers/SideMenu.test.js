import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import SideMenu from '../../../src/components/layers/SideMenu';

const mockStore = configureMockStore();
const store = mockStore();

describe('<SideMenu />', () => {
  const props = {
    classes: {
      toolbar: {},
      drawer: {},
      drawerOpen: {},
      drawerClose: {}
    },
    theme: {
      direction: 'ltl'
    }
  };
  it('renders the component', () => {
    const wrapper = shallow(<SideMenu store={store} classes={props.classes} theme={props.theme} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });
});
