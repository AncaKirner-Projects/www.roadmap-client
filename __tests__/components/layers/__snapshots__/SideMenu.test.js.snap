// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> renders the component 1`] = `
<WithStyles(Drawer)
  className=""
  classes={
    Object {
      "paper": "",
    }
  }
  open={true}
  variant="permanent"
>
  <div
    className={Object {}}
  >
    <WithStyles(IconButton)>
      <pure(ChevronLeftIcon) />
    </WithStyles(IconButton)>
  </div>
  <WithStyles(Divider) />
  <h4>
    Categories
  </h4>
  <WithStyles(List)>
    <Connect(Categories) />
  </WithStyles(List)>
</WithStyles(Drawer)>
`;
